docs/gameplay-matchmaker.md

```diff
@@ -1,182 +1,351 @@
+If the counts define a range, the matchmaker will try to return the max opponents possible but will never return less than the minimum count:
 
 ```js fct_label="JavaScript"
-// Updated example TBD
+const message = { matchmaker_add: {
+  min_count: 2,
+  max_count: 4,
+  query: "*"
+}};
+var ticket = await socket.send(message);
 ```
 
-```csharp fct_label=".Net"
-// Updated example TBD
+```csharp fct_label=".NET"
+var query = "*"
+var minCount = 2;
+var maxCount = 4;
+
+var matchmakerTicket = await socket.AddMatchmakerAsync(query, minCount, maxCount);
 ```
 
 ```csharp fct_label="Unity"
-// Updated example TBD
-```
+var query = "*"
+var minCount = 2;
+var maxCount = 4;
 
-### Using range filters
+var matchmakerTicket = await socket.AddMatchmakerAsync(query, minCount, maxCount);
+```
 
-Range filters operate on an inclusive lowerbound and an inclusive upperbound integer. The property should be within the two numbers for the matchmaking candidate to be considered.
+To search for an exact number of opponents submit the same minimum and maximum count:
 
 ```js fct_label="JavaScript"
-// Updated example TBD
+const message = { matchmaker_add: {
+  min_count: 4,
+  max_count: 4,
+  query: "*"
+}};
+var ticket = await socket.send(message);
 ```
 
-```csharp fct_label=".Net"
-// Updated example TBD
+```csharp fct_label=".NET"
+var query = "*"
+var minCount = 4;
+var maxCount = 4;
+
+var matchmakerTicket = await socket.AddMatchmakerAsync(query, minCount, maxCount);
 ```
 
 ```csharp fct_label="Unity"
-// Updated example TBD
+var query = "*"
+var minCount = 4;
+var maxCount = 4;
+
+var matchmakerTicket = await socket.AddMatchmakerAsync(query, minCount, maxCount);
 ```
 
-### Multi-filter matchmaking
+## Matchmaker tickets
 
-You can of course mix and match various filters to enhance the matchmaking search.
+Each time a user is added to the matchmaker pool they receive a ticket, a unique identifier for their entry into the pool.
 
 ```js fct_label="JavaScript"
-// Updated example TBD
+const message = { matchmaker_add: {
+  min_count: 2,
+  max_count: 4,
+  query: "*"
+}};
+var ticket = await socket.send(message);
 ```
 
-```csharp fct_label=".Net"
-// Updated example TBD
+```csharp fct_label=".NET"
+var query = "*"
+var minCount = 2;
+var maxCount = 4;
+
+var matchmakerTicket = await socket.AddMatchmakerAsync(query, minCount, maxCount);
 ```
 
 ```csharp fct_label="Unity"
-// Updated example TBD
+var query = "*"
+var minCount = 2;
+var maxCount = 4;
+
+var matchmakerTicket = await socket.AddMatchmakerAsync(query, minCount, maxCount);
 ```
 
-## Cancel a request
+This ticket is used to notify the client when matching completes to distinguish between multiple matchmaker operations. User may also cancel the matchmaking process using the ticket.
```

Comment:
> This ticket is used to notify the client when matching completes to distinguish between multiple matchmaker operations

This doesn't read well. The ticket isn't used to notify the client - I really tried to come up with something better here, but can't think of any - cc @novabyte